Description: Use system jemalloc
 Upstream seem to be "difficult" on this matter. See:
 https://github.com/antirez/redis/pull/137
Author: Chris Lamb <lamby@debian.org>

--- redis-2.4.2.orig/src/Makefile
+++ redis-2.4.2/src/Makefile
@@ -72,8 +72,7 @@
 
 ifeq ($(MALLOC),jemalloc)
-  DEPENDENCY_TARGETS+= jemalloc
-  FINAL_CFLAGS+= -DUSE_JEMALLOC -I../deps/jemalloc/include
-  FINAL_LIBS+= ../deps/jemalloc/lib/libjemalloc.a -ldl
+  FINAL_CFLAGS+= -DUSE_JEMALLOC -I/usr/include/jemalloc
+  FINAL_LIBS+= -ljemalloc
 endif
 
 REDIS_CC=$(QUIET_CC)$(CC) $(FINAL_CFLAGS)
--- redis-2.4.2.orig/src/zmalloc.c
+++ redis-2.4.2/src/zmalloc.c
@@ -61,10 +61,10 @@
 #define realloc(ptr,size) tc_realloc(ptr,size)
 #define free(ptr) tc_free(ptr)
 #elif defined(USE_JEMALLOC)
-#define malloc(size) je_malloc(size)
-#define calloc(count,size) je_calloc(count,size)
-#define realloc(ptr,size) je_realloc(ptr,size)
-#define free(ptr) je_free(ptr)
+#define malloc(size) malloc(size)
+#define calloc(count,size) calloc(count,size)
+#define realloc(ptr,size) realloc(ptr,size)
+#define free(ptr) free(ptr)
 #endif
 
 #ifdef HAVE_ATOMIC
--- a/src/zmalloc.h.orig	2012-10-23 14:56:21.272493961 +0100
+++ b/src/zmalloc.h	2012-10-23 14:56:29.008474117 +0100
@@ -50,7 +50,7 @@
 #include <jemalloc/jemalloc.h>
 #if (JEMALLOC_VERSION_MAJOR == 2 && JEMALLOC_VERSION_MINOR >= 1) || (JEMALLOC_VERSION_MAJOR > 2)
 #define HAVE_MALLOC_SIZE 1
-#define zmalloc_size(p) je_malloc_usable_size(p)
+#define zmalloc_size(p) malloc_usable_size(p)
 #else
 #error "Newer version of jemalloc required"
 #endif
